lib/logitech_receiver/hidpp20.py

```diff
@@ -328,13 +328,19 @@ def __getitem__(self, index):
 					if self.keyversion == 1:
 						self.keys[index] = _ReprogrammableKeyInfo(index, ctrl_id_text, ctrl_task_text, flags)
 					if self.keyversion == 4:
-						mapped_data = feature_request(self.device, FEATURE.REPROG_CONTROLS_V4, 0x20, key&0xff00, key&0xff)
```

Comment:
@doctor64 introduced this functionality in PR #337 FWIW

Reply:
Hmm, never see error here, but I don't have MX Anywhere 2 mouse. Anyway, error checking looks Ok, i'm little old school, and never trust in try/catch :) 
Out of curiosity, can you provide command trace of failed command?

Reply:
Looks like Logitech messed up with virtual buttons, to understand we need a traces

Reply:
That's the result of:
 $ python -m trace --trackcalls bin/solaar show

https://pastebin.com/4NA3vEvr

PS.: when trace is on, it doesn't print the error message:
solaar: error: {'params': '\x00\xc3', 'request': 2092L, 'number': 3, 'error': 2}

Reply:
@mchehab I mean command trace, output of solaar -dd show
Sorry for misunderstanding

Reply:
@doctor64 OK, you likely want just the last parts of the trace, after getting the 7th programable key, right?

There it is:

         6: RIGHT SCROLL AS AC PAN    , default: HorzScrollRightSet          => RIGHT SCROLL AS AC PAN    
             divertable, mse, reprogrammable, pos:0, group:3, gmask:15
22:17:26,415    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 0819 070000]
22:17:26,434    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 0819 00C3009C800004000100000000000000]
22:17:26,435    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082D 00C300]
22:17:26,454    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 FF08 2D020000000000000000000000000000]
22:17:26,454    ERROR [MainThread] logitech_receiver.base: (3) device 3 error on feature request {082D}: 2 = invalid argument
solaar: error: {'params': '\x00\xc3', 'request': 2093L, 'number': 3, 'error': 2}

Reply:
If you want, that's the full trace for the programmable keys:
```
     Has 8 reprogrammable keys:
22:17:26,132    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 081F 000000]
22:17:26,152    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 081F 00500038110001010000000000000000]
22:17:26,152    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082B 005000]
22:17:26,172    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082B 00500000000000000000000000000000]
         0: LEFT CLICK                , default: LeftClick                   => LEFT CLICK                
             mse, reprogrammable, pos:0, group:1, gmask:1
22:17:26,173    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 081E 010000]
22:17:26,192    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 081E 00510039110001010000000000000000]
22:17:26,192    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082A 005100]
22:17:26,212    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082A 00510000000000000000000000000000]
         1: RIGHT CLICK               , default: RightClick                  => RIGHT CLICK               
             mse, reprogrammable, pos:0, group:1, gmask:1
22:17:26,213    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 081C 020000]
22:17:26,232    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 081C 005200A93100020B0100000000000000]
22:17:26,232    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082F 005200]
22:17:26,252    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082F 00520000000000000000000000000000]
         2: MIDDLE BUTTON             , default: unknown:00A9                => MIDDLE BUTTON             
             divertable, mse, reprogrammable, pos:0, group:2, gmask:11
22:17:26,253    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 0819 030000]
22:17:26,274    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 0819 0053003C3100030F0100000000000000]
22:17:26,274    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082E 005300]
22:17:26,292    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082E 00530000000000000000000000000000]
         3: BACK AS BUTTON 4          , default: BackEx                      => BACK AS BUTTON 4          
             divertable, mse, reprogrammable, pos:0, group:3, gmask:15
22:17:26,293    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 081F 040000]
22:17:26,314    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 081F 0056003E3100030F0100000000000000]
22:17:26,314    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082F 005600]
22:17:26,334    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082F 00560000000000000000000000000000]
         4: FORWARD AS BUTTON 5       , default: BrowserForwardEx            => FORWARD AS BUTTON 5       
             divertable, mse, reprogrammable, pos:0, group:3, gmask:15
22:17:26,335    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 0819 050000]
22:17:26,354    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 0819 005B003F3100030F0100000000000000]
22:17:26,354    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082E 005B00]
22:17:26,374    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082E 005B0000000000000000000000000000]
         5: LEFT SCROLL AS AC PAN     , default: HorzScrollLeftSet           => LEFT SCROLL AS AC PAN     
             divertable, mse, reprogrammable, pos:0, group:3, gmask:15
22:17:26,375    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 081A 060000]
22:17:26,394    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 081A 005D00403100030F0100000000000000]
22:17:26,394    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082F 005D00]
22:17:26,414    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 082F 005D0000000000000000000000000000]
         6: RIGHT SCROLL AS AC PAN    , default: HorzScrollRightSet          => RIGHT SCROLL AS AC PAN    
             divertable, mse, reprogrammable, pos:0, group:3, gmask:15
22:17:26,415    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 0819 070000]
22:17:26,434    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 0819 00C3009C800004000100000000000000]
22:17:26,435    DEBUG [MainThread] logitech_receiver.base: (3) <= w[10 03 082D 00C300]
22:17:26,454    DEBUG [MainThread] logitech_receiver.base: (3) => r[11 03 FF08 2D020000000000000000000000000000]
22:17:26,454    ERROR [MainThread] logitech_receiver.base: (3) device 3 error on feature request {082D}: 2 = invalid argument
solaar: error: {'params': '\x00\xc3', 'request': 2093L, 'number': 3, 'error': 2}
```